### PR TITLE
[Keyboard] eHan Keyboard

### PR DESCRIPTION
**Pull request by suavity**
## Description
Creation of eHan F13 10.5° Split WKL, TKL Keyboard designed by Henry from Suavity Designs. The keyboard is similar to the FLX Virgo and LZ Ergo, with the exception that eHan's PCB offers 3U Spacebar Support, an F13 Key, and large Winkey blockers. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/suavity/ehan/config.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5344 // 5344 short for "SD" (Suavity Designs)
+#define PRODUCT_ID      0x4548 // 4548 short for "EH" (EHan)
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Suavity Designs
+#define PRODUCT         ehan
+#define DESCRIPTION     A 10.5° Split TKL Custom Keyboard
```
Comment (drashna) [inline keyboards/suavity/ehan/config.h:26]:
```suggestion
```

File: keyboards/suavity/ehan/readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# ehan: A 10.5° Split TKL Keyboard designed by Suavity Designs | QMK Firmware
+
+<p align="center">
+  <img align="middle" src="https://svd-image.b-cdn.net/Icons%20&%20SVG/eHan_etKL_icon_website.png"  width="400"> 
+</p>
```
Comment (drashna) [inline keyboards/suavity/ehan/readme.md:5]: Please use markdown here, not html.  

Also, please host on imgur.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/suavity/ehan/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+*/
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_ansi_default(
+        KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5,         KC_F6, KC_F7, KC_F8, KC_F9,     KC_F10, KC_F11,KC_F12, KC_F13,   KC_PSCR, KC_SLCK, KC_PAUS,  \
+        KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6,      KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSLS,  KC_BSPC,        KC_INS,  KC_HOME, KC_PGUP,  \
+        KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T,            KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS,           KC_DEL,  KC_END,  KC_PGDN,  \
+        KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G,            KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,      KC_ENT,                                       \
+        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B,       KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,      KC_RSFT,   MO(1),         KC_UP,            \
```
Comment (fauxpark) [inline keyboards/suavity/ehan/keymaps/default/keymap.c:25]:
```suggestion
        KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5,         KC_F6, KC_F7, KC_F8, KC_F9,     KC_F10, KC_F11,KC_F12, KC_F13,   KC_PSCR, KC_SLCK, KC_PAUS,
        KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6,      KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSLS,  KC_BSPC,        KC_INS,  KC_HOME, KC_PGUP,
        KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T,            KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS,           KC_DEL,  KC_END,  KC_PGDN,
        KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G,            KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,      KC_ENT,
        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B,       KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,      KC_RSFT,   MO(1),         KC_UP,
```

File: keyboards/suavity/ehan/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        KC_LCTL,           KC_LALT, KC_LCMD, KC_SPC,                 KC_SPC,      KC_RALT,      KC_RWIN,      KC_RCTL,      KC_LEFT, KC_DOWN, KC_RGHT),
+
+
+
+  [1] = LAYOUT_ansi_default(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,                                         \
+        KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,       KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,   KC_TRNS,           KC_TRNS,           \
```
Comment (fauxpark) [inline keyboards/suavity/ehan/keymaps/default/keymap.c:35]:
```suggestion
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,
        KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,       KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,   KC_TRNS,           KC_TRNS,
```

File: keyboards/suavity/ehan/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+*/
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+   [0] = LAYOUT_ansi_default(
+        KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5,         KC_F6, KC_F7, KC_F8, KC_F9,     KC_F10, KC_F11,KC_F12, KC_F13,   KC_PSCR, KC_SLCK, KC_PAUS,  \
+        KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6,      KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSLS,  KC_BSPC,        KC_INS,  KC_HOME, KC_PGUP,  \
+        KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T,            KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS,           KC_DEL,  KC_END,  KC_PGDN,  \
+        KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G,            KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,      KC_ENT,                                       \
+        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B,       KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,      KC_RSFT,   MO(1),         KC_UP,            \
```
Comment (fauxpark) [inline keyboards/suavity/ehan/keymaps/via/keymap.c:25]:
```suggestion
        KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5,         KC_F6, KC_F7, KC_F8, KC_F9,     KC_F10, KC_F11,KC_F12, KC_F13,   KC_PSCR, KC_SLCK, KC_PAUS,
        KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6,      KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSLS,  KC_BSPC,        KC_INS,  KC_HOME, KC_PGUP,
        KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T,            KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS,           KC_DEL,  KC_END,  KC_PGDN,
        KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G,            KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,      KC_ENT,
        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B,       KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,      KC_RSFT,   MO(1),         KC_UP,
```

File: keyboards/suavity/ehan/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        KC_LCTL,           KC_LALT, KC_LCMD, KC_SPC,                 KC_SPC,      KC_RALT,      KC_RWIN,      KC_RCTL,      KC_LEFT, KC_DOWN, KC_RGHT),
+
+
+   [1] = LAYOUT_ansi_default(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,                                         \
+        KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,       KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,   KC_TRNS,           KC_TRNS,           \
```
Comment (fauxpark) [inline keyboards/suavity/ehan/keymaps/via/keymap.c:34]:
```suggestion
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,
        KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,       KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,   KC_TRNS,           KC_TRNS,
```

File: keyboards/suavity/ehan/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        KC_TRNS,           KC_TRNS, KC_TRNS, KC_DEL,                 KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,                                  KC_TRNS, KC_TRNS, KC_TRNS),
+
+
+   [2] = LAYOUT_ansi_default(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,\
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,                                         \
+        KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,       KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,   KC_TRNS,           KC_TRNS,           \
```
Comment (fauxpark) [inline keyboards/suavity/ehan/keymaps/via/keymap.c:43]:
```suggestion
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,
        KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,       KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,   KC_TRNS,           KC_TRNS,
```

File: keyboards/suavity/ehan/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,                 KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,                                  KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [3] = LAYOUT_ansi_default(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,  \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,\
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,                                         \
+        KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,       KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,   KC_TRNS,           KC_TRNS,           \
```
Comment (fauxpark) [inline keyboards/suavity/ehan/keymaps/via/keymap.c:51]:
```suggestion
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS,         KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,
        KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,       KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,   KC_TRNS,           KC_TRNS,
```

File: keyboards/suavity/ehan/keymaps/via/rules.mk [hunk 9]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+BOOTMAGIC_ENABLE = lite
```
Comment (fauxpark) [inline keyboards/suavity/ehan/keymaps/via/rules.mk:3]:
```suggestion
```

File: keyboards/suavity/ehan/rules.mk [hunk 10]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = STM32F072
```
Comment (fauxpark) [inline keyboards/suavity/ehan/rules.mk:2]:
```suggestion
MCU = STM32F072

# Bootloader selection
BOOTLOADER = stm32-dfu
```

File: keyboards/suavity/ehan/rules.mk [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = full      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/suavity/ehan/rules.mk:12]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/suavity/hanjie/rules.mk [hunk 12]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = STM32F072
```
Comment (fauxpark) [inline keyboards/suavity/hanjie/rules.mk:2]:
```suggestion
MCU = STM32F072

# Bootloader selection
BOOTLOADER = stm32-dfu
```

File: keyboards/suavity/hanjie/rules.mk [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/suavity/hanjie/rules.mk:12]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by tzarc** [comment]
Why does this have two keyboards in it?

File: keyboards/suavity/ehan/info.json [hunk 14]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (noroadsleft) [inline keyboards/suavity/ehan/info.json:1]: This file's... not ideal... but it'll be easier to fix it once the PR is merged.

image

File: keyboards/suavity/ehan/readme.md [hunk 15]
```diff
@@ -0,0 +1,23 @@
+# ehan: A 10.5° Split TKL Keyboard designed by Suavity Designs | QMK Firmware
+
+![Imgur](https://imgur.com/bAiBbjg)
```
Comment (noroadsleft) [inline keyboards/suavity/ehan/readme.md:3]:
```suggestion
![Imgur](https://i.imgur.com/bAiBbjgh.png)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/suavity/ehan/config.h [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+#define PRODUCT_ID      0x4548 // 4548 short for "EH" (EHan)
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Suavity Designs
+#define PRODUCT         ehan
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 17
+
+#define MATRIX_COL_PINS { A10, A9, A8, B15, B14, B13, B12, C14, B7, B6, B5,  B4,  B3,  A15, C13, B9, B8, }
```
Comment (noroadsleft) [inline keyboards/suavity/ehan/config.h:31]:
```suggestion
#define MATRIX_COL_PINS { A10, A9, A8, B15, B14, B13, B12, C14, B7, B6, B5,  B4,  B3,  A15, C13, B9, B8 }
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by noroadsleft** [comment]
I'll fix the `info.json` issues in a follow-up PR.

Thanks!

**Review by noroadsleft — APPROVED** [review]

**Pull request merged**